### PR TITLE
fix: lack check metric type when no segment

### PR DESCRIPTION
**Pull request by smellthemoon**
#31970

### CODE REVIEW

**Comment by smellthemoon** [comment]
rerun ut

File: internal/proxy/task_search.go [hunk 1]
```diff
@@ -443,11 +445,39 @@ func (t *searchTask) tryGeneratePlan(params []*commonpb.KeyValuePair, dsl string
 		}
 		annsFieldName = vecFields[0].Name
 	}
-	queryInfo, offset, parseErr := parseSearchInfo(params, t.schema.CollectionSchema, ignoreOffset)
+
+	annField := typeutil.GetFieldByName(t.schema.CollectionSchema, annsFieldName)
+	listIndexResp, respErr := t.node.(*Proxy).dataCoord.ListIndexes(t.ctx, &indexpb.ListIndexesRequest{
```
Comment (czs007) [inline internal/proxy/task_search.go:450]: too weird

**Pull request closed without merging**